From dff28c4b9c21ec319f206be320fc33ba02b24cea Mon Sep 17 00:00:00 2001
From: David Reaver <me@davidreaver.com>
Date: Mon, 3 Feb 2025 13:13:09 -0800
Subject: [RFC PATCH 1/6] debugfs: Define debugfs_node to be dentry
To: Greg Kroah-Hartman <gregkh@linuxfoundation.org>,
    Rafael J. Wysocki <rafael@kernel.org>,
    Danilo Krummrich <dakr@kernel.org>
Cc: Steven Rostedt <rostedt@goodmis.org>,
    Christian Brauner <brauner@kernel.org>,
    Alexander Viro <viro@zeniv.linux.org.uk>,
    Tejun Heo <tj@kernel.org>,
    linux-fsdevel@vger.kernel.org,
    cocci@inria.fr,
    linux-kernel@vger.kernel.org

Add a #define debugfs_node dentry to support migrating all debugfs APIs
from dentry to a new opaque descriptor called debugfs_node. Subsequent
commits will apply a Coccinelle script and manual fixups to migrate the
rest of the kernel before replacing this #define with a struct definition.

This is done inside dcache.h so every user of dentry can see this #define.
If this was in debugfs.h we would have to add tons of imports, or find all
relevant forward defines of struct dentry; and add a struct debugfs_node;
next to them. This is temporary anyway and is removed in the last commit in
this series.

Signed-off-by: David Reaver <me@davidreaver.com>
---
 include/linux/dcache.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/dcache.h b/include/linux/dcache.h
index 4afb60365675..4b0c11cd3d50 100644
--- a/include/linux/dcache.h
+++ b/include/linux/dcache.h
@@ -127,6 +127,8 @@ struct dentry {
 	} d_u;
 };
 
+#define debugfs_node dentry
+
 /*
  * dentry->d_lock spinlock nesting subclasses:
  *
